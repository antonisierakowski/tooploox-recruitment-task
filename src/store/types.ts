import { GhUserState } from './ghUser/types';
import { NotificationsState } from './notifications/types';

export interface RootState {
  // ghUser: GhUserState;
  // notifications: NotificationsState
}

export interface Action<TPayload = any> {
  type: string;
  payload?: TPayload;
}
