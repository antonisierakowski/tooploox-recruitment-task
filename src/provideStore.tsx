import React from 'react';
import { Store } from 'redux';
import { Provider } from 'react-redux';

export const provideStore = (
  component: React.ReactElement,
  store: Store,
): React.ReactElement => <Provider store={store} children={component} />;
