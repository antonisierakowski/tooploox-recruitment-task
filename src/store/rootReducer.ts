import { combineReducers } from 'redux';
import { notificationsReducer } from './notifications/reducer';
import { ghUserReducer } from './ghUser/reducer';

export const rootReducer = combineReducers({
  ghUserState: ghUserReducer,
  notificationsState: notificationsReducer,
});
