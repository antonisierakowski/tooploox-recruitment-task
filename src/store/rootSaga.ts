import { all } from 'redux-saga/effects';
import { notificationSaga } from './notifications/sagas';

export function* rootSaga() {
  yield all([notificationSaga()]);
}
