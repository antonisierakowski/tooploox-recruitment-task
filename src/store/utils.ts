import { Action } from './types';

export function createAction<TPayload>(
  actionType: string,
  payload?: TPayload,
): Action<TPayload> {
  return {
    type: actionType,
    payload,
  };
}
