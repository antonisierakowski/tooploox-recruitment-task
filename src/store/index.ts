import { createStore, applyMiddleware, Store } from 'redux';
import { rootReducer } from './rootReducer';
import createSagaMiddleware from 'redux-saga';
import { composeWithDevTools } from 'redux-devtools-extension';
import { Action, RootState } from './types';
import { rootSaga } from './rootSaga';

const sagaMiddleware = createSagaMiddleware();

const middleware = composeWithDevTools(applyMiddleware(sagaMiddleware));

export const store: Store<RootState, Action> = createStore(
  rootReducer,
  middleware,
);

export const createPreloadedStore = (
  state: RootState,
): Store<RootState, Action> => createStore(rootReducer, state, middleware);

sagaMiddleware.run(rootSaga);
