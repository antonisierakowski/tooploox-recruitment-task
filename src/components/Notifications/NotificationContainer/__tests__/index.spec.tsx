import { createPreloadedStore } from '../../../../store';
import { rootStateFixture } from '../../../../fixtures/rootStateFixture';
import { render, RenderResult, fireEvent } from '@testing-library/react';
import React from 'react';
import { NotificationContainer } from '../index';
import { notificationStateFixture } from '../../../../fixtures/notifiationStateFixture';
import { provideStore } from '../../../../provideStore';

const store = createPreloadedStore(rootStateFixture);

let renderResult: RenderResult;

beforeEach(() => {
  renderResult = render(provideStore(<NotificationContainer />, store));
});

it('should match to the generated snapshot', () => {
  expect(renderResult.asFragment()).toMatchSnapshot();
});

it('should render notifications', () => {
  for (const notification of notificationStateFixture.notifications) {
    const notificationComponent = renderResult.getByTestId(notification.id);
    expect(notificationComponent).toBeTruthy();
  }
});

it('should close notifications', async () => {
  const { id } = notificationStateFixture.notifications[0];

  expect(renderResult.getByTestId(id)).toBeInTheDocument();

  const closeIcon = renderResult.getByTestId(`closeIcon_${id}`);
  fireEvent.click(closeIcon);

  expect(renderResult.queryByTestId(id)).toBeNull();
});
