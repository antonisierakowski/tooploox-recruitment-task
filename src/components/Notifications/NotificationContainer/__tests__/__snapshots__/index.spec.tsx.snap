// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match to the generated snapshot 1`] = `
<DocumentFragment>
  <div
    class="notificationContainer"
  >
    <div
      class="notification"
      data-testid="1"
    >
      <div
        class="notificationBody"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          data-testid="successIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
          />
        </svg>
        <p
          class="notificationText"
        >
          test1
        </p>
      </div>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root closeIcon"
        data-testid="closeIcon_1"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </div>
    <div
      class="notification"
      data-testid="2"
    >
      <div
        class="notificationBody"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          data-testid="failureIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
          />
        </svg>
        <p
          class="notificationText"
        >
          test2
        </p>
      </div>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root closeIcon"
        data-testid="closeIcon_2"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </div>
    <div
      class="notification"
      data-testid="3"
    >
      <div
        class="notificationBody"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          data-testid="failureIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
          />
        </svg>
        <p
          class="notificationText"
        >
          test3
        </p>
      </div>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root closeIcon"
        data-testid="closeIcon_3"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
